executor/new_distsql.go

```diff
@@ -300,10 +300,52 @@ type IndexLookUpExecutor struct {
 	tableWorker
 	finished chan struct{}
 
-	resultChan <-chan *lookupTableTask
+	resultCh   *resultChannel
 	resultCurr *lookupTableTask
 }
 
+type resultChannel struct {
+	keepOrder  bool
+	taskCh     chan *lookupTableTask
+	curID      int
+	waitChLock sync.Mutex
+	waitChMap  map[int]chan struct{}
+}
+
+func (c *resultChannel) fetchTask() *lookupTableTask {
+	task := <-c.taskCh
+	if task != nil {
```

Comment:
Why check task is not nil and do nothing?

Reply:
Redundant code..